contracts/registry/PocketRegistry.sol

```diff
@@ -54,36 +45,37 @@ contract PocketRegistry is NodeCrud {
   // By registering a Node, you are agreeing to be a relayer in the Pocket Network.
   // Three actions happen - you burn some PKT, register in the registry, and a Node contract gets created and assigned to your address
   // Registry allows network to keep track of current live nodes
-  function registerNode(address _nodeAddress, string8[] _supportedTokens, string _url, uint8 _port, uint _index) {
-    require(delegateContract.delegatecall(bytes4(sha3("registerNode()")), _nodeAddress, _supportedTokens, _url, _port, _index));
+  function registerNode(address _nodeAddress, string8[] _supportedTokens, string _url, uint8 _port, bool _isRelayer, bool _isOracle) onlyOwner {
+    require(delegateContract.delegatecall(bytes4(sha3("registerNode()")), _nodeAddress, _supportedTokens, _url, _port, _isRelayer, _isOracle));
   }
 
-  function createNodeContract(string8[] _supportedTokens, string _url, uint8 _port, bool _isRelayer, bool _isOracle) {
+  // This is the function to create a new Node.
+  function createNodeContract(string8[] _supportedTokens, string _url, uint8 _port, bool _isRelayer, bool _isOracle) onlyOwner{
     require(delegateContract.delegatecall(bytes4(sha3("createNodeContract()")), _supportedTokens, _url, _port, _isRelayer, _isOracle));
   }
 
-  // Updates the values of the given Node record.
-  function updateNode(address _nodeAddress, string url) {
-    require(delegateContract.delegatecall(bytes4(sha3("updateNode()")), _nodeAddress, url));
+  // Updates the values of the given Registered Node record.
+  function updateNodeRecord(address _nodeAddress, string8[] _supportedTokens, string _url, uint8 _port, bool _isRelayer, bool _isOracle) onlyOwner {
+    require(delegateContract.delegatecall(bytes4(sha3("updateNodeRecord()")), _nodeAddress, _supportedTokens, _url, _port, _isRelayer, _isOracle));
   }
 
   // Unregister a given Node record
-  function unregisterNode(address _nodeAddress) {
-    require(delegateContract.delegatecall(bytes4(sha3("unregisterNode()")), _nodeAddress));
+  function unregisterNodeRecord(address _nodeAddress) onlyOwner {
+    require(delegateContract.delegatecall(bytes4(sha3("unregisterNodeRecord()")), _nodeAddress));
   }
 
   // Transfer ownership of a given record.
-  function transfer(address _nodeAddress, address newOwner) {
-    require(delegateContract.delegatecall(bytes4(sha3("transfer()")), _nodeAddress, newOwner));
+  function transferNode(address _nodeAddress, address newOwner) onlyOwner {
+    require(delegateContract.delegatecall(bytes4(sha3("transferNode()")), _nodeAddress, newOwner));
```

Comment:
Delegate calls need the full signature passed in as string (including parameters), also we should keep these function names in a global so we don't have to hash it every time the method is called.